### ISSUE TITLE
New criterion: attachment size

### ISSUE THREAD

**Issue by gellnerm** [eyalroz/removedupes#128]
Thank you for this addon. 

My use case is saving space on my mail account. However I have many duplicated mails that differ in any other criterion except the sum of the attachment sizes. Other mails that match one of the implemented criteria are not abundant and not so huge.

Can we have an attachment size criterion?

**Comment by gellnerm** [comment]
> you have messages with the same content and headers, and attachments with the same name, but different sizes?

No, they have different content and headers. But attachment names and sizes are equal.
These messages are created e.g. when forwarding mails to different recipients and adding some comments.

**Comment by gellnerm** [comment]
> Should you not use the headers (and/or the number of lines of the content)?

Because they cannot be used as criterion if they are both inequal

**Comment by eyalroz** [comment]
Ah, I see now. Well, let's put it this way: I occasionally get a request to add a niche criterion which someone wants. My standard policy on these is: If a few people - not just one - asks for this feature, I'll implement it. At the moment I find this not likely to be of interest to many people.

**Comment by gellnerm** [comment]
Ok. Let me list some functions that may be helpful for implementation.

Get attachments of a message:
https://webextension-api.thunderbird.net/en/102/messages.html#listattachments-messageid

Each attachment has a size attribute:
https://webextension-api.thunderbird.net/en/102/messages.html#messageattachment

(Probably not needed) Get attachment file object:
https://webextension-api.thunderbird.net/en/102/messages.html#getattachmentfile-messageid-partname

```
let attachments = await browser.messages.listAttachments(messageId);
let size_sum = 0;
for (let att of attachments) {
    let size = att.size;
    size_sum = size_sum + size;
    ...
}
```

**Comment by eyalroz** [comment]
Your links are to WebExtensions API. Mine is not a webextension, I work on the same level as TB's own chrome. Anyway, the implementation for these APIs may (or may not) be extremely slow - removing most of the benefit of using it.
